thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/alert/scheme/DetectionJiraAlerter.java

```diff
@@ -76,43 +78,81 @@ public DetectionJiraAlerter(ADContentFormatterContext adContext, ThirdEyeAnomaly
+    }
+
+    Properties jiraClientConfig = new Properties();
+    jiraClientConfig.putAll(ConfigUtils.getMap(notificationSchemeProps.get(PROP_JIRA_SCHEME)));
+
+    List<AnomalyResult> anomalyResultListOfGroup = new ArrayList<>(anomalies);
+    anomalyResultListOfGroup.sort(COMPARATOR_DESC);
+
+    BaseNotificationContent content = super.buildNotificationContent(jiraClientConfig);
+    return new JiraContentFormatter(this.jiraAdminConfig, jiraClientConfig, content,
+        this.teConfig, adContext).getJiraEntity(anomalyResultListOfGroup);
   }
 
   private void createJiraTickets(DetectionAlertFilterResult results) {
     LOG.info("Preparing a jira alert for subscription group id {}", this.adContext.getNotificationConfig().getId());
     Preconditions.checkNotNull(results.getResult());
     for (Map.Entry<DetectionAlertFilterNotification, Set<MergedAnomalyResultDTO>> result : results.getResult().entrySet()) {
       try {
-        Map<String, Object> notificationSchemeProps = result.getKey().getNotificationSchemeProps();
-        if (notificationSchemeProps == null || notificationSchemeProps.get(PROP_JIRA_SCHEME) == null) {
-          throw new IllegalArgumentException("Invalid jira settings in subscription group " + this.adContext.getNotificationConfig().getId());
+        JiraEntity jiraEntity = buildJiraEntity(result.getKey(), result.getValue());
+        Iterable<Issue> issues = jiraClient.getIssue(jiraEntity.getJiraProject(), jiraEntity.getSummary(),
+            this.jiraAdminConfig.getJiraUser()).getIssues();
+
+        List<Issue> jiraIssues = new ArrayList<>();
+        if (issues != null) {
+          // Sort the issues by descending order of create time
+          jiraIssues = StreamSupport.stream(issues.spliterator(), false).sorted(new Comparator<Issue>() {
+            @Override
+            public int compare(Issue o1, Issue o2) {
+              return o2.getCreationDate().compareTo(o1.getCreationDate());
+            }
+          }).collect(Collectors.toList());
         }
 
-        Properties jiraClientConfig = new Properties();
-        jiraClientConfig.putAll(ConfigUtils.getMap(notificationSchemeProps.get(PROP_JIRA_SCHEME)));
-
-        List<AnomalyResult> anomalyResultListOfGroup = new ArrayList<>(result.getValue());
-        anomalyResultListOfGroup.sort(COMPARATOR_DESC);
-
-        BaseNotificationContent content = super.buildNotificationContent(jiraClientConfig);
-        IssueInput jiraIssueInput = new JiraContentFormatter(this.jiraAdminConfig, jiraClientConfig, content, this.teConfig, adContext)
-            .getJiraEntity(anomalyResultListOfGroup);
-
-        String issueKey = createIssue(jiraIssueInput);
-        LOG.info("Jira ticket created with issue key {} : anomalies reported = {}", issueKey, result.getValue().size());
+        String issueKey;
+        if (jiraIssues.isEmpty() || jiraIssues.get(0).getCreationDate().isBefore(System.currentTimeMillis() - jiraEntity.getMergeGap())) {
```

Comment:
Should the `mergeGap` come from the config or the `jiraEntity`? I assume the `mergeGap` is the same within one subscription group.

Reply:
Since mergeGap applies only to jira alerts (email and iris are not mergeable), I kept it as a jira property. Maybe I can rename it to `reopenTimeGap` for clarity?